### ISSUE TITLE
Interval not reached, even when car is charging

### ISSUE THREAD

**Issue by wopfel** [db-EV/ZoePHP#22]
Hi,
The non-charging interval is set to 60 minutes, the charging interval is set to 15 minutes in my configuration.

My car is charging now, and 'Charging status' is correctly set to 1 in database.csv file. For this, I've forced the data gathering by setting `$cron_ncs` to a very low value.

After waiting for another 15 minutes, the `php index.php cron` is still reporting `INTERVAL NOT REACHED`.

I've added some debugging output, and it seems `$session[6]` is still set to `N` (which should be `Y` since the car is charging). 

I guess the issue is in https://github.com/db-EV/ZoePHP/blob/9aa3d6b5e54460295c1787940862623d21cd8400/src/index.php#L338-L345

`if ($session[10] == 1) $session[6] = 'Y';` (line 344) is only executed if the above if-clause (line 338) is true. Shouldn't `$session[6]` be set to `Y` regardless of the mail/exec setting? (Maybe moving lines 344+345 out of the if-clause?)

From my `config.php` file:
```
$mail_csf = 'N';
$exec_csf = '';
```

Regards,
Bernd

**Comment by db-EV** [comment]
Yes, this seems to be a bug if you use cron only for database saving and not for mail as well.
But the solution should be to replace line 75 with this one I think:
`if ($session[10] == 1 || $session[6] == 'Y') date_add($s, date_interval_create_from_date_string($cron_acs.' minutes'));`

**Comment by wopfel** [comment]
Thanks for your reply. I‘ll test it tomorrow when the sun‘s shining and let you know.

Reminder for me:
https://github.com/db-EV/ZoePHP/blob/9aa3d6b5e54460295c1787940862623d21cd8400/src/index.php#L75

**Comment by db-EV** [comment]
It doesn't update the session[6] value but this value is only used for the mail and exec function to detect when charging is finished.
For only database saving the session[6] value doesn't matter. Therefore I think row 75 changing above should fix the issue.
Awaiting your test resultat before publishing a new version.

**Comment by wopfel** [comment]
Changing line 75 seems to be sufficient, you're right. I've tested it right now.
+1 for the change you suggested.

Details:

```
% php index.php cron
INTERVAL NOT REACHED: 202205081329. Session6=N, Session10=1.%
% date
So 8. Mai 14:30:50 CEST 2022
% vim index.php
% php index.php cron
INTERVAL NOT REACHED: 202205081244. Session6=N, Session10=1.%
```

Explanation of output (some debugging output was added by me):
- Car is charging, data was fetched.
- Running script again (1st line): 60 minutes were added (which is wrong, since car is charging)
- date command shows current time
- Changed line 75 (vim command)
- Running modified script, the interval is calculated correctly (15 minutes added)
